lib/ansible/modules/cloud/cloudscale/cloudscale_volume.py

```diff
@@ -146,77 +166,86 @@ class AnsibleCloudscaleVolume(AnsibleCloudscaleBase):
 
     def __init__(self, module):
         super(AnsibleCloudscaleVolume, self).__init__(module)
-        params = self._module.params
-        self.info = {
-            'name': params['name'],
-            'uuid': params['uuid'],
+        self._info = {}
+
+    def _init_container(self):
+        return {
+            'uuid': self._module.params.get('uuid') or self._info.get('uuid'),
+            'name': self._module.params.get('name') or self._info.get('name'),
             'state': 'absent',
         }
-        self.changed = False
-        if params['uuid'] is not None:
-            vol = self._get('volumes/%s' % params['uuid'])
-            if vol is None and params['state'] == 'present':
-                self._module.fail_json(
-                    msg='Volume with UUID %s does not exist. Can\'t create a '
-                        'volume with a predefined UUID.' % params['uuid'],
```

Comment:
This case is not covered by the new code. It will probably just throw an exception at line 213.